pkg/hubble/server/server.go

```diff
@@ -68,13 +65,23 @@ func (s *Server) initGRPCServer() {
 // listeners. Stop should be called to stop the server.
 func (s *Server) Serve() error {
 	s.initGRPCServer()
-	for name, listener := range s.opts.Listeners {
-		go func(name string, listener net.Listener) {
-			s.log.WithField("listener", name).Info("Starting gRPC server on listener")
-			if err := s.srv.Serve(listener); err != nil {
-				s.log.WithError(err).Error("failed to close grpc server")
+	if s.opts.UnixSocketListener != nil {
+		go func() {
+			if err := s.srv.Serve(s.opts.UnixSocketListener); err != nil {
+				s.log.WithError(err).
+					WithField("address", s.opts.UnixSocketListener.Addr().String()).
+					Error("failed to start grpc server")
+			}
+		}()
+	}
+	if s.opts.TCPListener != nil {
+		go func() {
+			if err := s.srv.Serve(s.opts.TCPListener); err != nil {
+				s.log.WithError(err).
+					WithField("address", s.opts.TCPListener.Addr().String()).
+					Error("failed to start grpc server")
```

Comment:
nit: to avoid code duplication, you could keep the loop.
```go
	for _, listener := range []net.Listener{s.opts.UnixSocketListener, s.opts.TCPListener} {
		if listener != nil {
			go func(listener net.Listener) {
				if err := s.srv.Serve(listener); err != nil {
					s.log.WithError(err).
						WithField("address", listener.Addr().String()).
						Error("failed to start grpc server")
				}
			}(listener)
		}
	}
```

Reply:
fixed